setup.go

```diff
@@ -1,29 +1,43 @@
 package main
 
 import (
-  "fmt"
-  "net/http"
-  "log"
-  "io/ioutil"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
 )
 
-func main(){
-  var url string
-  fmt.Println(">> Insert url: (Make sure to use http://)")
-  fmt.Scanf("%s", &url)
-  if url == "" {
-    fmt.Println("Please type in an url")
-  }
-  data, err := http.Get(url)
-  if err != nil{
-    log.Fatal(err)
-  }else{
-    fmt.Println("Fetching url...")
-    defer data.Body.Close()
-    contents, err := ioutil.ReadAll(data.Body)
-    if err != nil{
-      log.Fatal(err)
-    }
-    fmt.Println("%s\n", string(contents))
-  }
+// askURL provide an i/o process asking for a valid URL
+func askURL() string {
+	var url string
+	fmt.Print("URL: ")
+	fmt.Scanf("%s", &url)
+
+	if (url == "") ||
+		((!strings.HasPrefix(url, "http:")) ||
+			(!strings.HasPrefix(url, "http:"))) {
```

Comment:
sorry, this line should be `(!strings.HasPrefix(url, "https:"))) {` instead of `(!strings.HasPrefix(url, "http:"))) {` :laughing: 

can you fix it @christoga ?